### PR TITLE
Add auto-extending of names

### LINKED ISSUE #491: Auto extend names
Display a checkbox next to each name to auto-extend when possible.

Next to each name user should be able to mark for auto-extend.

### PR DESCRIPTION
**Pull request by etharner**
Closes #491.

### CODE REVIEW

File: src/store/plugins/names.js [hunk 1]
```diff
@@ -72,8 +76,12 @@ export default store =>
                 { root: true },
               );
             }
+            if (name.expiresAt <= height && oldName.autoExtend) {
```
Comment (mradkov) [inline src/store/plugins/names.js:79]: I think the logic here is wrong. 

If the `height` is gt than `name.expiresAt` then the name would have been already expired. 🤔 

Maybe just auto extend without checking the height?

Comment (mradkov) [inline src/store/plugins/names.js:79]:
```suggestion
            if (name.expiresAt >= height && oldName.autoExtend) {
```

This way we will ensure we dispatch updatePointer only in the case when the current block height has not reached the name's expiration date.

**Review by mradkov — CHANGES_REQUESTED** [review]
Also while testing - the name extend popup appears every few seconds when I'm on the `Names` page.

Let's make it fire only once at opening the extension.

File: src/popup/utils/constants.js [hunk 2]
```diff
@@ -107,6 +107,7 @@ export const MAX_AMOUNT_WITHOUT_CONFIRM = 10;
 export const AENS_DOMAIN = '.chain';
 export const MAX_AUCTION_NAME_LENGTH = 12 + AENS_DOMAIN.length;
 export const MIN_NAME_LENGTH = 14;
+export const AUTO_EXTEND_NAME_BLOCKS_INTERVAL = 100;
```
Comment (etharner) [inline src/popup/utils/constants.js:110]: Let's maybe use some interval between blocks when we will be asking for extending, and I also call it only on extension opening

**Review by mradkov — APPROVED** [review]

**Pull request merged**